Added note about list DOM fields and GQI (#2869)
SkylineCommunications/dataminer-docs: user-guide/Advanced_Modules/DOM/DOM_objects/DOM_SectionDefinition.md

```

> [!NOTE]
>
> - From DataMiner 10.4.2/10.5.0 onwards, it is no longer possible to pass empty lists as value for a `FieldDescriptor` that allows multiple values if that field is required. A `FieldDescriptor` that is not required will still allow empty lists as value, but note that it is best practice to not pass values for `FieldDescriptors` if the value is empty or an empty list.

> > - Fields with multiple values, like fields used in multiple sections, will not be available in GQI queries.

Reviewer: The current phrasing could be a little ambiguous. Do we need to mention the bit about "fields used in multiple sections" or can we just leave this out here? If it's not necessary, removing it is a simple way to clear any possible confusion. Otherwise, I'd suggest rephrasing this:
```suggestion
> - Similar to fields used in multiple sections, fields with multiple values are also not available in GQI queries.
```

Author: This refers to the feature where you can allow multiple sections, thus multiple values, for a SectionDefinition on a DOM instance by setting the 'AllowMultipleSections' bool on the SectionDefinitionLink (or status link). (https://docs.dataminer.services/user-guide/Advanced_Modules/DOM/DOM_objects/DomDefinition.html#sectiondefinitionlink)

We could also remove it here, but we probably should also add something about the fact that the multiple section fields also won't be available in GQI. Not sure where this would fit best.

Author: Yes, sounds great to me.
